### PR TITLE
Fix / update linux.scp action #1236

### LINKED ISSUE #1236: Fix / update linux.scp action
Currently, `linux.scp` action is mostly useless if you want to download files from a remote server to a local one since it expects a `dest_server` argument.

We have a couple of ways to solve this:
1. Make `scp` action "desination agnostic" - instead of taking `dest_server` parameter, make it just take `source` and `destination` arguments. This way it works both ways (upload, download) since source and destination can either be a remote host with a path or just a path.
2. Rename existing `scp` action to `scp_upload` and add a new inverse `scp_download` action.

### PR DESCRIPTION
**Pull request by nmaludy**
PR to fix #1236

I liked Kami's approach 1). In my mind this is more flexible because the author using the ```linux.scp``` action may not know if this will be an "upload" or a "download" or potentially a "transfer" (```scp user@host1:a.txt user@host2:```).

By using ``` source``` and ```destination```  this accounts of all scenarios and allows for each to be accomplished using the same action. Also, this mimics the way that a user would interact with ```scp``` on the command line.

Now, i know this breaks backwards compatibility. Is this OK, or should i mark ```dest_server``` as  ```required: false``` for the time being and some Jinja logic to insert it in the command if it's specified?

Thoughts?

### CODE REVIEW

**Comment by estee-tew** [comment]
https://stackstorm.slack.com/archives/community/p1491431996752869

**Comment by Kami** [comment]
Thanks for this change.

I'm fine with it as long as we document this "breaking" change in CHANGELOG.rst and in upgrade notes in st2docs repo (https://github.com/StackStorm/st2docs/blob/master/docs/source/upgrade_notes.rst).

File: contrib/linux/actions/scp.yaml [hunk 1]
```diff
@@ -13,10 +13,6 @@
             type: 'string'
             description: 'Destination of files/directories'
```
Comment (Kami) [inline contrib/linux/actions/scp.yaml]: Let's please add an example to the description to make it clear destination can be relative, absolute, or full server:path notation.

Comment (Kami) [inline contrib/linux/actions/scp.yaml]: @nmaludy Can you please look into this change and the one mentioned in changelog so we can get it merged once that is resolved?

Thanks :)

**Comment by nmaludy** [comment]
Thanks for the comments. Should be updated now based on your requests.

**Comment by Kami** [comment]
Related st2docs PR - https://github.com/StackStorm/st2docs/pull/500

We also need to update our build servers, etc. before we can merge this because it's a backward incompatible change.

**Comment by Kami** [comment]
As mentioned in https://github.com/StackStorm/st2docs/pull/500#issuecomment-307728838, I decided to make this change backward compatible (so it will work using either approach for a next release or two) because otherwise merging it would be quite messy and we would need to update a lot of our CI / CD related code.

**Comment by Kami** [comment]
Let's continue in https://github.com/StackStorm/st2/pull/3463.

Thanks again.

**Pull request closed without merging**